cvat/apps/engine/views.py

```diff
@@ -2105,7 +2112,84 @@ def content(self, request, pk):
-            return Response(data=manifest_files, content_type="text/plain")
+            return Response(
+                data=manifest_files,
+                content_type='text/plain',
+                headers={'Deprecation': datetime(2023, 7, 1, tzinfo=timezone.get_current_timezone()).strftime('%a, %d %B %Y %H:%M:%S %Z')}
+            )
+
+        except CloudStorageModel.DoesNotExist:
+            message = f"Storage {pk} does not exist"
+            slogger.glob.error(message)
+            return HttpResponseNotFound(message)
+        except (ValidationError, PermissionDenied, NotFound) as ex:
+            msg = str(ex) if not isinstance(ex, ValidationError) else \
+                '\n'.join([str(d) for d in ex.detail])
+            slogger.cloud_storage[pk].info(msg)
+            return Response(data=msg, status=ex.status_code)
+        except Exception as ex:
+            slogger.glob.error(str(ex))
+            return Response("An internal error has occurred",
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    @extend_schema(summary='Method returns the content of the cloud storage',
+        parameters=[
+            OpenApiParameter('manifest_path', description='Path to the manifest file in a cloud storage',
+                location=OpenApiParameter.QUERY, type=OpenApiTypes.STR),
+            OpenApiParameter('prefix', description='Prefix to filter data',
+                location=OpenApiParameter.QUERY, type=OpenApiTypes.STR),
+            OpenApiParameter('delimiter', description='The character you use to group keys in the cloud storage',
+                location=OpenApiParameter.QUERY, type=OpenApiTypes.STR),
+            OpenApiParameter('next_token', description='Used to continue listing files in the bucket',
+                location=OpenApiParameter.QUERY, type=OpenApiTypes.STR),
+            OpenApiParameter('page_size', location=OpenApiParameter.QUERY, type=OpenApiTypes.INT),
+        ],
+        responses={
+            '200': OpenApiResponse(response=CloudStorageContentSerializer, description='A manifest content'),
+        },
+    )
+    @action(detail=True, methods=['GET'], url_path='content-v2')
+    def content_v2(self, request, pk):
+        storage = None
+        try:
+            db_storage = self.get_object()
+            storage = db_storage_to_storage_instance(db_storage)
+            prefix = request.query_params.get('prefix')
+            page_size = request.query_params.get('page_size', str(settings.BUCKET_CONTENT_MAX_PAGE_SIZE))
+            if not page_size.isnumeric():
+                return HttpResponseBadRequest('Wrong value for page_size was found')
+            page_size = min(int(page_size), settings.BUCKET_CONTENT_MAX_PAGE_SIZE)
+
+            # make api identical to share api
+            if prefix and prefix.startswith('/'):
+                prefix = prefix[1:]
+            delimiter = request.query_params.get('delimiter', '/')
```

Comment:
Do we actually need this parameter?

First, I don't see it being used anywhere in the UI.

Second, I think we would need extensive tests for a case where a storage bucket uses a delimiter other than '/', as I expect that a lot of things might break in this case (basically, any path manipulation code related to cloud paths is liable to break). I don't see any such tests right now.

Third, even if we do decide to support such a case, I think it would make more sense for the delimiter to be a property of the cloud storage object, rather than an operation parameter.